ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -462,10 +462,62 @@ private void expandNestedListType(String jsonFieldName, Object val, Field field,
     });
   }
 
+  /**
+   * This method gives us a chance to perform some additional type coercions in case the BigQuery
+   * field type is different from the source data type. This should rarely happen, since only
+   * validated payloads get through to this BQ sink path, but there are sets of probes with
+   * heterogeneous types that appear as explicit fields in BQ, but are treated as loosely typed
+   * maps at the validation phase; we need to catch these or they can cause the entire pipeline
+   * to stall.
+   *
+   * <p>Returning {@code null} here indicates that no coercion is defined and that the field should
+   * be put to {@code additional_properties}.
+   */
+  private static Optional<Object> coerceToBqType(Object o, Field field) {
+    if (field.getMode() == Mode.REPEATED) {
+      if (o instanceof List) {
+        return Optional.of(((List<Object>) o).stream().map(v -> coerceSingleValueToBqType(v, field))
+            // We have not yet observed a case where an array type contains values that cannot
+            // be coerced to appropriate values, so this filter should not be getting rid of
+            // elements; a future improvement would be to refactor so that we can insert
+            // non-coerceable values from a list into additional_properties.
+            .filter(Optional::isPresent).map(Optional::get).collect(Collectors.toList()));
```

Comment:
We now coerce objects within a list here. This handles the case of JSON-ifying items in a list that was previously done in `processField`.